transformations/butter_fingers_perturbation_for_IL/transformation.py

```diff
@@ -0,0 +1,105 @@
+import os
+import itertools
+import random
+import json
+
+from interfaces.SentenceOperation import SentenceOperation
+from tasks.TaskTypes import TaskType
+
+'''
+    This transformation adds noise to all types of text sources (sentence, paragraph, etc.), which is proportional to noise erupting from keyboard typos resulting in common spelling errors. 
+    We have expanded the existing implementation for English (source) to a few Indian Languages:
+    1. Bangla ('bn'), 
+    2. Gujarati ('gu'), 
+    3. Hindi ('hi'), 
+    4. Kannada ('kn'), 
+    5. Malayalam ('ml'), 
+    6. Oriya ('or'), 
+    7. Punjabi ('pa'), 
+    8. Tamil ('ta'), 
+    9. Telugu ('te')
+
+    args:
+        language (str): base language for butter fingers
+        keyboard (str): preffered keyboard schema (currently available: "inscript")
+        seed (int, default = 0): seed for reproducibility 
+'''
+
+def butter_finger(text, language:str, keyboard:str, prob=0.1,  seed=0, max_outputs=1):
+
+    mapping_path = os.path.join(os.path.dirname(os.path.abspath(__file__)), 'key_mapping.json')
+
+    with open(mapping_path,'r') as f:
+        key_mapping = json.load(f)
+
+    languages = list(key_mapping.keys())
+
+    if language not in key_mapping:
+        raise Exception(f"Language not supported. Available languages: {languages}.")
+
+    keyboards = list(key_mapping[language].keys())
+    if keyboard not in key_mapping[language]:
+        raise Exception(f"Keyboard not supported. Available keyboards for '{language}': {keyboards}.")
+
+    random.seed(seed)
+    key_approx = key_mapping[language][keyboard]
+
+    prob_of_typo = int(prob * 100)
+    perturbed_texts = []
+    for _ in itertools.repeat(None, max_outputs):
+        butter_text = ""
+        for letter in text:
+            if letter not in key_approx.keys():
+                new_letter = letter
+            else:
+                if random.choice(range(0, 100)) <= prob_of_typo:
+                    new_letter = random.choice(key_approx[letter])
+                else:
+                    new_letter = letter
+
+            butter_text += new_letter
+        perturbed_texts.append(butter_text)
+    return perturbed_texts
+
+
+"""
+Butter Finger implementation borrowed and edited from https://github.com/alexyorke/butter-fingers.
+"""
+
+
+class ButterFingersPerturbation(SentenceOperation):
+    tasks = [
+        TaskType.TEXT_CLASSIFICATION,
+        TaskType.TEXT_TO_TEXT_GENERATION,
+        TaskType.TEXT_TAGGING,
+    ]
```

Comment:
In addition to the tasks, add a `keywords` list following the instructions in the [keywords](https://github.com/GEM-benchmark/NL-Augmenter/blob/main/docs/keywords.md) documentation.

Also add a `languages` list with only the two letter code to comply with the standard of all the other transformations.